Entity/Dynamic.php

```diff
@@ -170,6 +170,10 @@ public function getData()
      */
     public function __set($name, $value)
     {
+        if ($value instanceof \DateTime) {
```

Comment:
save it here as datetime (update the orm file)  and fix the __get function when its from type `date` to convert it back to a dateTime.

So add something like this to #L208

```
if (strpos($name, 'date') === 0) {
   return new \DateTime($array[$name]);
}
```

Test some edge cases.